skbio/io/format/genbank.py

```diff
@@ -0,0 +1,669 @@
+
+
+genbank = create_format('genbank')
+
+# date format in locus line of genbank record
+_TIME_FORMAT = '%d-%b-%Y'
+# This list is ordered
+# used to read and write genbank file.
+_HEADERS = ['LOCUS',
+            'DEFINITION',
+            'ACCESSION',
+            'VERSION',
+            'DBSOURCE',
+            'DBLINK',
+            'KEYWORDS',
+            'SOURCE',
+            'REFERENCE',
+            'COMMENT',
+            'FEATURES',
+            'ORIGIN']
+
+
+@genbank.sniffer()
+def _genbank_sniffer(fh):
+    # check the 1st real line is a valid LOCUS line
+    if _too_many_blanks(fh, 5):
+        return False, {}
+    try:
+        line = next(_line_generator(fh, skip_blanks=True, strip=False))
+    except StopIteration:
+        return False, {}
+
+    try:
+        _parse_locus([line])
+    except GenbankFormatError:
+        return False, {}
+    return True, {}
+
+
+@genbank.reader(None)
+def _genbank_to_generator(fh, constructor=None, **kwargs):
+    for record in _parse_genbanks(fh):
+        yield _construct(record, constructor, **kwargs)
+
+
+@genbank.reader(Sequence)
+def _genbank_to_biological_sequence(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), seq_num)
+    return _construct(record, Sequence, **kwargs)
+
+
+@genbank.reader(DNA)
+def _genbank_to_dna(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), seq_num)
+    return _construct(record, DNA, **kwargs)
+
+
+@genbank.reader(RNA)
+def _genbank_to_rna(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), seq_num)
+    return _construct(record, RNA, **kwargs)
+
+
+@genbank.reader(Protein)
+def _genbank_to_protein(fh, seq_num=1, **kwargs):
+    record = _get_nth_sequence(_parse_genbanks(fh), seq_num)
+    return _construct(record, Protein, **kwargs)
+
+
+@genbank.writer(None)
+def _generator_to_genbank(obj, fh, lowercase=None):
+    for obj_i in obj:
+        _serialize_single_genbank(obj_i, fh, lowercase)
+
+
+@genbank.writer(Sequence)
+def _biological_sequence_to_genbank(obj, fh, lowercase=None):
+    _serialize_single_genbank(obj, fh, lowercase)
+
+
+@genbank.writer(DNA)
+def _dna_to_genbank(obj, fh, lowercase=None):
+    _serialize_single_genbank(obj, fh, lowercase)
+
+
+@genbank.writer(RNA)
+def _rna_to_genbank(obj, fh, lowercase=None):
+    _serialize_single_genbank(obj, fh, lowercase)
+
+
+@genbank.writer(Protein)
+def _protein_to_genbank(obj, fh, lowercase=None):
+    _serialize_single_genbank(obj, fh, lowercase)
+
+
+def _construct(record, constructor=None, **kwargs):
+    seq, md, pmd = record
+    if 'lowercase' not in kwargs:
+        kwargs['lowercase'] = True
+    if constructor is None:
+        unit = md['LOCUS']['unit']
+        if unit == 'bp':
+            is_rna = 'u' in seq or 'U' in seq
+            # often RNA mol type has T instead of U for genbank from from NCBI;
+            # but in case of files from other sources...
```

Comment:
> @jairideout it would be unsatisfying, as molecule type is free text and necessarily have "DNA" or "RNA" in it.

Thanks, that makes sense. The format description makes it sound like this is a controlled vocabulary but from looking at the examples it appears it isn't :(